### PR TITLE
Make spotbugs-annotations provided and enforce non-runtime

### PR DESCRIPTION
**Pull request by robobario**
### Type of change

- Bugfix

### Description

This dependency is build time only and doesn't need to be present on the runtime.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

**Comment by robobario** [comment]
note that spotbugs isn't in the 0.7.0 distribution libs dir because the scope was managed to `provided` in kroxylicious-app, I guess it preferred the scope from dependencyManagement in the parent.

```
robeyoun@robeyoun-thinkpadp16vgen1:~/development/upstream/kroxylicious/kroxylicious-app$ mvn dependency:tree -Dverbose | grep spotbugs -B10
...
[INFO] |  +- io.apicurio:apicurio-registry-schema-validation-jsonschema:jar:0.0.7:runtime (version managed from 0.0.7)
...
[INFO] |  +- com.github.spotbugs:spotbugs-annotations:jar:4.8.6:provided (version managed from 4.8.6; scope managed from compile)
```

File: kroxylicious-filters/kroxylicious-record-validation/pom.xml [hunk 1]
```diff
@@ -111,7 +111,7 @@
         <dependency>
             <groupId>com.github.spotbugs</groupId>
             <artifactId>spotbugs-annotations</artifactId>
-            <scope>compile</scope>
+            <scope>provided</scope>
```
Comment (SamBarker) [inline kroxylicious-filters/kroxylicious-record-validation/pom.xml:114]: If the scope is set in the dependency management section to provided would we be better removing scope here?

**Review by SamBarker — APPROVED** [review]
Good call banning it from those scopes.

File: kroxylicious-filters/kroxylicious-record-validation/pom.xml (hunk 1 above)
Comment (robobario) [reply kroxylicious-filters/kroxylicious-record-validation/pom.xml]: done, I've removed all the redundant overrides

**Pull request merged**